pkg/operator/resources/namespaced/controller.go

```diff
@@ -128,6 +129,16 @@ func createControllerDeployment(controllerImage, importerImage, clonerImage, upl
 		deployment.Spec.Template.Spec.PriorityClassName = priorityClassName
 	}
 	container := utils.CreateContainer("cdi-controller", controllerImage, verbosity, pullPolicy)
+	container.Ports = []corev1.ContainerPort{
+		{
+			Name:          "metrics",
+			ContainerPort: 8080,
+			Protocol:      "TCP",
+		},
+	}
+	labels := util.MergeLabels(deployment.Spec.Template.GetLabels(), map[string]string{common.PrometheusLabelKey: common.PrometheusLabelValue})
+	deployment.SetLabels(labels)
+	deployment.Spec.Template.SetLabels(labels)
```

Comment:
Isn't this going to break the deployment if we upgrade? Since the template is immutable.

Reply:
`spec.selector` is immutable, spec.template.metadata.labels/deployment.metadata.labels are fine
(otherwise k8s basically says you can't add labels to this resource at all)